### ISSUE TITLE
[css-fonts-3] Default feature list should not require a list of features to turn on

### ISSUE THREAD

**Issue by litherum** [w3c/csswg-drafts#1267]
https://drafts.csswg.org/css-fonts-3/#default-features

The set of default features should be able to be different on different platforms. Following platforms' text conventions is valuable.

**Comment by litherum** [comment]
cc @svgeesus

**Comment by svgeesus** [comment]
The Opentype feature registry uses should, not must. See for example
https://www.microsoft.com/typography/otspec/features_ko.htm#locl
"UI suggestion: This feature should be active by default."

**Comment by svgeesus** [comment]
This is on the agenda for CSS WG telcon 13 Dec 2017. If we keep the must, I volunteer to write the tests for that section. But we would also want to put it at-risk because I don't think we have interop there. If we drop down to may, it becomes untestable (except for the 'switching off' clause). I would like to get this section updated either way, for an early 2018 updated CR,

**Comment by dbaron** [comment]
Is the proposal to add the allowance that *additional* features may be enabled by default (which seems reasonable to me), or are you proposing that some or all of the features currently required by that section not be required (I'd be more skeptical of this change, although maybe there are some specific ones that could change)?

**Comment by svgeesus** [comment]
I committed some tests for this section (each has several subtests). I believe these test everything that this section mandates. Here are the results:
https://test.csswg.org/harness/results/css-fonts-3_dev/grouped/section/7.1/

Firefox and Chrome pass all tests, so we do have interop there. Safari passes most subtests but not all, which does lead to failing each test it t's entirety. Edge does not currently implement this so naturally fails all the tests.

I agree with David that allowing implementations to exceed the requirements (can do better) is noble and also testable and workable. Watering down the spec to `may` is unfortunate for content developers, because they then have to set a bunch of stuff just to get interop between browsers.

**Comment by dbaron** [comment]
@litherum: https://github.com/w3c/csswg-drafts/issues/1267#issuecomment-356485302 is a question for you, really

**Comment by svgeesus** [comment]
I was asked to test on more platforms so we now have passes on:

Chrome 56.0.2924.87 (Blink 537.36) on Android 6.0.1
Chrome 63.0.3239.84 (Blink 537.36) on Mac OS X Intel 10.13.1
Chrome 65.0.3315.4 (Blink 537.36) on Windows NT 10.0
Firefox 52.0 (Gecko 20100101) on Linux i686
Firefox 58.0 (Gecko 58.0) on Android 6.0.1
Safari 11.0.2 (WebKit 604.4.7) on Mac OS X Intel 10.11.6

fails on
Safari 11.0.1 (WebKit 604.3.5) on Mac OS X Intel 10.13.1
Microsoft Edge 16.16299 (Edge 16.16299) on Windows NT 10.0
https://test.csswg.org/harness/results/css-fonts-3_dev/grouped/section/7.1/

iOS still to-do

**Comment by litherum** [comment]
The macOS port of WebKit (and presumably the iOS port too, though I haven't checked) doesn't enable the `mkmk` feature by default, because the rest of the macOS platform also doesn't enable it by default. It shouldn't be required in CSS.

We also don't enable `vert` in vertical text, but this is a bug.

These are the only two failures the macOS port of WebKit displays.

**Comment by khaledhosny** [comment]
Not enabling `mkmk` feature by default looks like a bug and would break mark positioning in many fonts.

**Comment by svgeesus** [comment]
Myles, thanks for getting back on this. I agree that Webkit is very nearly compliant; Fonts 3 requires **all** of the listed features to be enabled by default and thus the test I wrote requires all of them, rather than testing each individually. If I had done the unit tests, Webkit would be passing most of them.

Yes, not enabling `vert` in vertical text is certainly a bug. Firefox had the same bug, which I reported and they fixed.

Khaled, I agree. This will certainly break Vietnamese and Arabic (for the latter see [Developing OpenType Fonts for Arabic Script (3 of 3): Features](https://www.microsoft.com/typography/otfntdev/arabicot/features.aspx) ) but even for what Microsoft charmingly calls "standard scripts" (Latin, Cyrillic, Greek and Armenian) `mkmk` is listed as **required**:
[Developing OpenType Fonts for Standard Scripts](https://www.microsoft.com/typography/OpenTypeDev/standard/intro.htm)
